text/0000-ember-data-packages.md

```diff
@@ -0,0 +1,396 @@
+      <td>import { BooleanTransform } from 'ember-data/transforms';</td>
+    </tr>
+    <tr>
+      <td colspan="3"><h3>@ember-data/store</h3></td>
+    </tr>
+    <tr>
+      <td>DS.Store</td>
+      <td>import Store from 'ember-data/store';</td>
+      <td>import Store from '@ember-data/store';</td>
+    </tr>
+    <tr>
+      <td>DS.Snapshot</td>
+      <td>none</td>
+      <td>none</td>
+    </tr>
+    <tr>
+      <td>DS.PromiseArray</td>
+      <td>none</td>
+      <td>none</td>
+    </tr>
+    <tr>
+      <td>DS.PromiseObject</td>
+      <td>none</td>
+      <td>none</td>
+    </tr>
+    <tr>
+      <td>DS.RecordArray</td>
+      <td>none</td>
+      <td>none</td>
+    </tr>
+    <tr>
+      <td>DS.AdapterPopulatedRecordArray</td>
+      <td>none</td>
+      <td>none</td>
+    </tr>
+    <tr>
+      <td>DS.RecordarrayManager</td>
+      <td>none</td>
+      <td>none</td>
+    </tr>
+    <tr>
+      <td>DS._initializeStoreService</td>
+      <td>import initializeStoreService from 'ember-data/intialize-store-service';</td>
+      <td>none</td>
+    </tr>
+    <tr>
+      <td>DS._setupContainer</td>
+      <td>import setupContainer from 'ember-data/setup-container';</td>
+      <td>none</td>
+    </tr>
+    <tr>
+      <td>DS.normalizeModelName</td>
+      <td>none</td>
+      <td>import { normalizeModelName } from 'ember-data/store';<br>
+        <br>this public method should be a candidate for deprecation
+      </td>
+    </tr>
+    <tr>
+      <td colspan="3"><h3>@ember-data/relationship-layer</h3></td>
+    </tr>
+    <tr>
+      <td>DS.Relationship</td>
+      <td>none</td>
+      <td>none</td>
+    </tr>
+    <tr>
+      <td colspan="3"><h3>@ember-data/debug</h3></td>
+    </tr>
+    <tr>
+      <td>DS.DebugAdapter</td>
+      <td>none</td>
+      <td>none</td>
+    </tr>
+  </tbody>
+</table>
+
+### Notes
+
+#### `@ember-data/model`
+  
+  1) `InternalModel` and `RootState` are tightly coupled to the store and to our provided `Model`
+    implementation. Overtime we need to uncouple this, but given their coupling to `Model` and our
+    desire to enable them to be eliminated from projects not using `Model`, I believe these exports
+    belong in `@ember-data/model` for this reason.
+
+  2) Do the following belong in `@ember-data/model`, or in `@ember-data/relationship-layer`?
+    if not with relationships, does the package name "relationship-layer" become confusing?
+    
+   The argument for them being here is they are far more related to the current `Model` ui
+    presentation layer than they are to the state book-keeping which could be used with
+    alternative implementations.
+
+  * `belongsTo`
+  * `hasMany`
+  * `PromiseManyArray`
+  * `ManyArray`
+  
+#### `@ember-data/serializers`
+
+  1) We should move automatic registration of transforms into a more traditional
+    `app/` directory re-export for the package so that when the package is dropped they
+    cleanly drop as well.
+
+#### `@ember-data/store`
```

Comment:
Is there specific reasoning for why some of the package names are singular (e.g. `model`, `store`) and some are plural (`adapters`, `serializers`)?

Reply:
Yes. `adapters/` and `serializers/` are going to each contain ALL of the current adapter and serializer implementations, not just a single one. It's not practical for these implementations to be sectioned off by format (`json-api` `rest` `json` `base`). I would expect alternatives to these packages to be available in the future.